File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if(s.empty()) {
            return true;
        }

        string temp;

        for(auto& a: s) {
            if(isalnum(a)) {
                temp.push_back(tolower(a));
            }
        }

        for(int left = 0, right = temp.size() - 1; left < right; left++, right--) {
            if(temp[left] != temp[right]) {
                return false;
            }
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        if not s:
            return True

        temp = ''
        for char in s:
            if char.isalnum():
                temp += char.lower()
                
        left = 0
        right = len(temp) - 1
        while left < right:
            if temp[left] != temp[right]:
                return False
            left += 1
            right -= 1

        return True